techmatters/webchat
# Limit the characters in the webchat for specific fields

## Description
This PR adds the ability to apply HTML mutations to the form inputs provided by Twilio in the pre engagement form. ~I know this might be an overkill for now, as we've only been asked limit the length of certain inputs, so I'm happy to downgrade this to a simpler version~ Did it :)

Implemented adding listeners to the DOM via [MutationOberver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver), so the HTML changes are always applied even if the components are re-mounted (e.g. when the form is toggled to hidden).

This PR adds some example config on how it will be to add `maxLength` attribute to the an input called `firstName`. This changes will be reverted.
The `maxLength` attribute can be applied to the input definition itself (thanks @stephenhand for the feedback and the idea!).

### Checklist



### Verification steps
- `npm ci` & `npm run build:dev` & `npm start`.
- Confirm that the `firstName` input is limited to 30 characters.
